PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = ['crm.team']
-    _inherit = ['pos.config']
-    _inherit = ['pos.order']
-    _inherit = ['pos.order.line']
-    _inherit = ['pos.session']
-    _inherit = ['res.config.settings']
-    _inherit = ['stock.picking']

Total Changes: 16

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_sale/models/crm_team.py
+++ b/addons/pos_sale/models/crm_team.py
@@ -8,7 +8,7 @@ import pytz
 
 
 class CrmTeam(models.Model):
-    _inherit = ['crm.team']
+    _inherit = 'crm.team'
 
     pos_config_ids = fields.One2many('pos.config', 'crm_team_id', string="Point of Sales")
     pos_sessions_open_count = fields.Integer(string='Open POS Sessions', compute='_compute_pos_sessions_open_count')

--- a/addons/pos_sale/models/pos_config.py
+++ b/addons/pos_sale/models/pos_config.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     crm_team_id = fields.Many2one(
         'crm.team', string="Sales Team", ondelete="set null",

--- a/addons/pos_sale/models/pos_order.py
+++ b/addons/pos_sale/models/pos_order.py
@@ -6,7 +6,7 @@ from odoo.tools import float_compare, float_is_zero, format_date
 
 
 class PosOrder(models.Model):
-    _inherit = ['pos.order']
+    _inherit = 'pos.order'
 
     currency_rate = fields.Float(compute='_compute_currency_rate', store=True, digits=0, readonly=True)
     crm_team_id = fields.Many2one('crm.team', string="Sales Team", ondelete="set null")
@@ -170,7 +170,7 @@ class PosOrder(models.Model):
 
 
 class PosOrderLine(models.Model):
-    _inherit = ['pos.order.line']
+    _inherit = 'pos.order.line'
 
     sale_order_origin_id = fields.Many2one('sale.order', string="Linked Sale Order")
     sale_order_line_id = fields.Many2one('sale.order.line', string="Source Sale Order Line")

--- a/addons/pos_sale/models/pos_session.py
+++ b/addons/pos_sale/models/pos_session.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class PosSession(models.Model):
-    _inherit = ['pos.session']
+    _inherit = 'pos.session'
 
     crm_team_id = fields.Many2one('crm.team', related='config_id.crm_team_id', string="Sales Team", readonly=True)
 

--- a/addons/pos_sale/models/res_config_settings.py
+++ b/addons/pos_sale/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     pos_crm_team_id = fields.Many2one(related='pos_config_id.crm_team_id', readonly=False, string='Sales Team (PoS)')
     pos_down_payment_product_id = fields.Many2one(related='pos_config_id.down_payment_product_id', readonly=False)

--- a/addons/pos_sale/models/sale_order.py
+++ b/addons/pos_sale/models/sale_order.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models, _
 
 
 class SaleOrder(models.Model):
+    _name = 'sale.order'
     _inherit = ['sale.order', 'pos.load.mixin']
 
     pos_order_line_ids = fields.One2many('pos.order.line', 'sale_order_origin_id', string="Order lines Transfered to Point of Sale", readonly=True, groups="point_of_sale.group_pos_user")
@@ -60,6 +61,7 @@ class SaleOrder(models.Model):
 
 
 class SaleOrderLine(models.Model):
+    _name = 'sale.order.line'
     _inherit = ['sale.order.line', 'pos.load.mixin']
 
     pos_order_line_ids = fields.One2many('pos.order.line', 'sale_order_line_id', string="Order lines Transfered to Point of Sale", readonly=True, groups="point_of_sale.group_pos_user")

--- a/addons/pos_sale/models/stock_picking.py
+++ b/addons/pos_sale/models/stock_picking.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockPicking(models.Model):
-    _inherit = ['stock.picking']
+    _inherit = 'stock.picking'
 
     def _create_move_from_pos_order_lines(self, lines):
         lines_to_unreserve = self.env['pos.order.line']
